### PR TITLE
Basic Admin Functionality

### PR DESCRIPTION
**Pull request by abenetka**
- Adds enums to user model for default & admin users
- Updates DB for enum
- Adds admin/users controller with index action
- Admin dashboard page shows index of users
- All tests passing
- Driver/Navigator with @le3ah

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
+
   def create
-    if user = User.from_google_auth(request.env["omniauth.auth"])
+    user = User.from_google_auth(request.env["omniauth.auth"])
+    if user && user.role == "default"
       session[:user_id] = user.id
       flash[:success] = "Welcome, gardener, we're glad to have you among the roses."
       redirect_to dashboard_path
+    else
+      user && user.role == "admin"
```
Comment (asmolentzov) [inline app/controllers/sessions_controller.rb]: should this be `elsif user && user.role == "admin"`? Else seems like line 11 doesn't really do anything - just evaluates to true/false, but doesn't affect any other code.

File: spec/factories/users.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 FactoryBot.define do
   factory :user do
-    first_name { "MyString" }
-    last_name { "MyString" }
-    email { "MyString" }
+    first_name { "Gob" }
+    last_name { "Bluth" }
+    email { "gobis@bluth.com" }
```
Comment (asmolentzov) [inline spec/factories/users.rb]: 😂 👍

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -1,14 +1,20 @@
 class SessionsController < ApplicationController
   skip_before_action :require_login
-  
+
   def create
-    if user = User.from_google_auth(request.env["omniauth.auth"])
+    user = User.from_google_auth(request.env["omniauth.auth"])
+    if user && user.role == "default"
```
Comment (asmolentzov) [inline app/controllers/sessions_controller.rb]: This and the logic check below could probably both be helper methods eg `is_default_user` `is_admin_user` or similar. Eventual possible refactor.

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (bendelonlee) [reply app/controllers/sessions_controller.rb]: good catch @asmolentzov

Comment (abenetka) [reply app/controllers/sessions_controller.rb]: Totally agree! I'm going to make the changes now

**Pull request merged**